extensions/crossmodel-lang/src/language-server/cross-model-validator.ts

```diff
@@ -53,14 +55,7 @@ export class CrossModelValidator {
    }
 
    protected shouldHaveId(node: AstNode): boolean {
-      return (
-         isEntity(node) ||
-         isEntityAttribute(node) ||
-         isRelationship(node) ||
-         isSystemDiagram(node) ||
-         isDiagramEdge(node) ||
```

Comment:
I added the mapping but removed edges and nodes since they are not actually exported, i.e., externally visible. I cleaned up the whole validation a bit in a follow up commit to make that a little bit more clear.

Reply:
@martin-fleck-at So I can point to a relationship from another object? Like in the mapping in a join condition we can optionally reference to the id of a relationship right?

Or is that unrelated to this piece of code?

Reply:
In short: Yes, we expose relationships and also allow them to be referenced in the mapping.

Anything that we put into the global (as opposed to the document-local) index is in theory reference-able from outside a document and therefore also should have a unique ID (this validator). What is then actually available for referencing depends on the scope that we give (by default local and then global scope). However, this code is really just for showing an error for non-unique IDs.

Reply:
Ok, but shouldn't we validate whether the relationship id's are unique?

Reply:
We do, the `isExported` returns true for relationships :-)